src/common/transformations/src/transformations/transpose_sinking/ts_unsqueeze.cpp

```diff
@@ -178,6 +178,14 @@ TSUnsqueezeBackward::TSUnsqueezeBackward() {
             return false;
         }
 
+        // if main_node does nothing, just remove it
+        if (main_node->get_input_partial_shape(0) == main_node->get_output_partial_shape(0)) {
```

Comment:
is it correct when both `input` and `output` shapes are dynamic?

Reply:
In the perspective we would like to move TSGeneral transformation to the MOC, then NopElimination would work and the order of things would be more natural.